src/Components/Slider/Slider.js

```diff
@@ -0,0 +1,158 @@
+import React, { useState, useEffect, useRef } from 'react';
+import styled from 'styled-components';
+import { SliderContainer } from './SliderContainer';
+import { Slide } from './Slide';
+import { Arrow } from './Arrow';
+import { Dots } from './Dots';
+
+const SliderCSS = styled.div`
+  position: relative;
+  height: 100vh;
+  width: 100vw;
+  margin: 0 auto;
+  overflow: hidden;
+  white-space: nowrap;
+`
+
+const getWidth = () => window.innerWidth
+
+/**
+ * @function Slider
+ */
+const Slider = props => {
+
+  const slides  = Object.values(props);
+
+  const firstSlide = slides[0];
+  const secondSlide = slides[1];
+  const lastSlide = slides[slides.length - 1];
+
+  const [state, setState] = useState({
+    activeSlide: 0,
+    translate: getWidth(),
+    transition: 0.45,
+    transitioning: false,
+    _slides: [lastSlide, firstSlide, secondSlide]
+  });
+
+  const { activeSlide, translate, _slides, transition, transitioning } = state;
+
+  const autoPlayRef = useRef();
+  const transitionRef = useRef();
+  const resizeRef = useRef();
+  const sliderRef = useRef();
+  const throttleRef = useRef();
+
+
+  useEffect(() => {
+    autoPlayRef.current = nextSlide
+    transitionRef.current = smoothTransition
+    resizeRef.current = handleResize
+    throttleRef.current = throttleArrows
```

Comment:
It is recommended that the definition of the functions are done before they are used. This avoids confusion over globals or imported values.